### ISSUE TITLE
Accessibility support

### ISSUE THREAD

**Issue by Saulis** [vaadin/vaadin-text-field#48]
For reference: https://www.w3.org/TR/wai-aria-1.1/#textbox

Proposal: should support announcing label by the screenreader, also should announce states like disabled, readonly, invalid and required (and error message).

Also the value should be announced :doh:. And should announce that the user is in a textbox.

All the announcements should work out of the box without requiring the users to add any aria attributes themselves. `aria-label` can of course be used to override the default announcement.

Should be tested on VoiceOver (OSX + iOS), ChromeVox, JAWS and NVDA on Windows 10, Android TalkBack

**Comment by Saulis** [comment]
Notes from grooming: 
- try to limit the scope of this ticket so that no changes are required for the existing error message functionality.
